File: tests/python/arrays/count_squares/test_count_squares.py
from problems.arrays.count_squares.python.count_squares import countSquares
from tests.python.utility import load_test_cases


def test_count_squares() -> None:
    test_cases = load_test_cases("arrays/count_squares.json")

    for idx, test in enumerate(test_cases):
        points: list[list[int]] = test["points"]
        expected: int = test["expected"]

        result: int = countSquares(points)

        assert (
            result == expected
        ), f"Test case {idx} failed: expected {expected}, got {result}"

File: tests/cpp/arrays/test_count_squares.cpp
#include "../../../problems/arrays/count_squares/cpp/count_squares.h"
#include "../jsontestbase.h"

#include <vector>

class CountSquaresTest : public JsonTestBase {
public:
  CountSquaresTest() {
    json_file_path = std::string(TEST_CASES_DIR) + "/arrays/count_squares.json";
  }
};

TEST_F(CountSquaresTest, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty or not loaded";

  for (const auto &test : test_cases) {
    auto points = test["points"].get<std::vector<std::vector<int>>>();
    int expected = test["expected"].get<int>();

    auto result = countSquares(points);

    EXPECT_EQ(result, expected) << "Test case failed : " << test.dump(2);
  }
}
